src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +39,396 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
     (ActorFunc)BgSpot18Basket_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7710.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7770.s")
+static ColliderJntSphItemInit sJntSphItemsInit[2] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+        { 0, { { 0, 2040, 0 }, 54 }, 100 },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { 1, { { 0, 1400, 0 }, 13 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x09, 0x20, COLSHAPE_JNTSPH },
+    2,
+    sJntSphItemsInit,
+};
+
+static s16 D_808B85C8[] = { 32768, 10922, 54613, 0 };
+
+void func_808B7710(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+
+    Collider_InitJntSph(globalCtx, &this->colliderJntSph);
+    Collider_SetJntSph(globalCtx, &this->colliderJntSph, &this->dyna.actor, &sJntSphInit, &this->colliderJntSphItem);
+    this->dyna.actor.colChkInfo.mass = 255;
+}
+
+void func_808B7770(BgSpot18Basket* this, GlobalContext* globalCtx, f32 arg2) {
+    Vec3f acceleration;
+    Vec3f velocity;
+    Vec3f position;
+    f32 cosValue;
+    s32 i = 0;
+    f32 randomValue;
+    static s16 D_808B85D0 = 0;
+    f32 sinValue;
+    s32 arraySize = 2;
+
+    while (i != arraySize) {
+        if (globalCtx) {}
+        if (!(arg2 < Math_Rand_ZeroOne())) {
+            D_808B85D0 += 30000;
+
+            sinValue = Math_Sins(D_808B85D0);
+            cosValue = Math_Coss(D_808B85D0);
+
+            randomValue = (Math_Rand_ZeroOne() * 35.0f) + 35.0f;
+
+            position.x = (randomValue * sinValue) + this->dyna.actor.posRot.pos.x;
+            position.y = this->dyna.actor.posRot.pos.y + 10.0f;
+            position.z = (randomValue * cosValue) + this->dyna.actor.posRot.pos.z;
+
+            velocity.x = sinValue;
+            velocity.y = 0.0f;
+            velocity.z = cosValue;
+
+            acceleration.x = 0.0f;
+            acceleration.y = 0.5f;
+            acceleration.z = 0.0f;
+
+            func_800286CC(globalCtx, &position, &velocity, &acceleration, ((Math_Rand_ZeroOne() * 16) + 80),
+                          ((Math_Rand_ZeroOne() * 30) + 80));
+        }
+        i++;
+    }
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static s16 D_808B85E4[] = { 61536, 800, 4000 };
+
+void BgSpot18Basket_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+    Actor* actor = &this->dyna.actor;
+    s32 moveFlag = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, 3);
+    func_808B7710(this, globalCtx);
+    DynaPolyInfo_Alloc(&D_06002154, &moveFlag);
```

Comment:
`moveFlag` is not at all what this second parameter is for, just leave it unnamed as other actors